### PR TITLE
closes #252: Add err messages on client validation

### LINKED ISSUE #252: Form Field Messages
When any field has an error, use ng-messages to show a meaningful message to the user.

### PR DESCRIPTION
**Pull request by RyanBard**
* Adds descriptive error messages to UI for organization validation errors.
* Adds descriptive error messages to UI for program validation errors.
* Adds max length validation to UI for program street address.
* Adds pattern validation to UI for program zip code.
* Adds missing server validation to city/startyear/goal/etc.
* Adds config for common validation settings between server and client.
* Fixes indention in program-form-dialog.html.
* Adds comments and organization to application.properties.

### CODE REVIEW

File: config/application.properties [hunk 1]
```diff
@@ -1,25 +1,84 @@
...
+# Begin: Actuator endpoints config
+#####################################################################
+
+# I'm disabling these until we have an admin login to protect these with (they show sensitive things: db password)
 endpoints.env.enabled = false
 endpoints.configprops.enabled = false
 endpoints.logfile.enabled = false
 endpoints.heapdump.enabled = false
 
-logging.path = logs/
```
Comment (RyanBard) [inline config/application.properties]: This was moved to the top of the file.

File: src/main/java/org/hmhb/config/DefaultConfigService.java [hunk 2]
```diff
@@ -30,7 +30,12 @@ public DefaultConfigService(
 
         this.publicConfig = new PublicConfig(
                 environment.getProperty("google.oauth.client.id"),
-                environment.getProperty("hmhb.url.prefix")
+                environment.getProperty("hmhb.url.prefix"),
+                environment.getProperty("hmhb.program.startYear.minValue", Integer.class),
```
Comment (RyanBard) [inline src/main/java/org/hmhb/config/DefaultConfigService.java]: If you don't see it yet, you'll know if you review my tests why I'm uncomfortable with the pattern I have set...

File: src/main/java/org/hmhb/program/DefaultProgramService.java [hunk 3]
```diff
@@ -274,10 +288,64 @@ public Program save(
             throw new ProgramPrimaryGoal1RequiredException();
         }
 
+        if (program.getPrimaryGoal1().length() > publicConfig.getProgramGoalMaxLength()) {
```
Comment (RyanBard) [inline src/main/java/org/hmhb/program/DefaultProgramService.java]: This service is getting pretty dense.  A refactor is probably in order, but that can happen later -- at least for now, I have unit tests, so we can refactor and have confidence we're not totally screwing everything up.

File: src/test/java/org/hmhb/authentication/DefaultAuthenticationServiceTest.java [hunk 4]
```diff
@@ -64,7 +64,7 @@ private GoogleOauthAccessRequestInfo prepGoogleCall() throws Exception {
                 gPlusProfile
         );
 
-        PublicConfig publicConfig = new PublicConfig(CLIENT_ID, null);
+        PublicConfig publicConfig = new PublicConfig(CLIENT_ID, null, 1, 2, 3, 4, 5);
```
Comment (RyanBard) [inline src/test/java/org/hmhb/authentication/DefaultAuthenticationServiceTest.java]: Notice how the details of this object's constructor spill out all throughout the tests.

I should probably write a Builder for the config classes or something, but not in this PR.

**Comment by RyanBard** [comment]
I started to create screenshots, but there were alot of fields and permutations (and different browsers), so instead, you can interact with the form here: https://hmhb-ryan.herokuapp.com/views/map?time=1478138180135

**Comment by RyanBard** [comment]
Also, working on this spawned off #256 and #260 for followup work.

File: client/src/main/webjar/organization-form/organization-form.html [hunk 5]
```diff
@@ -59,6 +68,9 @@
             <input name="organizationFacebook"
                    ng-model="organizationForm.organization.facebookUrl"
                    type="url">
+            <div ng-messages="organizationForm.orgForm.organizationFacebook.$error">
+                <div ng-message="url">Organization Facebook page must be a valid url</div>
```
Comment (mckoon) [inline client/src/main/webjar/organization-form/organization-form.html]: We can come back to these later, but maybe we should mention what a valid URL looks like "start with https://" or something.

Comment (RyanBard) [inline client/src/main/webjar/organization-form/organization-form.html]: Yea, I started to do that with zip code (12345, 12345-1234, or 12345 1234), but I was concerned about the real estate -- too wordy.

**Pull request merged**